### PR TITLE
[Spec] update specs of aptos_governance

### PR DESCRIPTION
**Pull request by UIZorrot**
### Description

This is a part of the spec work for Aptos Framework from MoveBit. The changes are:

Completed most of the TODO in `aptos_governance.move`

There are many changes in this PR.

### Test Plan

`aptos move prove -f aptos_gov`

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/aptos_governance.spec.move [hunk 1]
```diff
@@ -149,18 +144,63 @@ spec aptos_framework::aptos_governance {
...
-        let stake_pool_res = global<stake::StakePool>(stake_pool);
-        aborts_if !exists<staking_config::StakingConfig>(@aptos_framework);
         aborts_if !exists<stake::StakePool>(stake_pool);
-        aborts_if global<stake::StakePool>(stake_pool).delegated_voter != proposer_address;
+        aborts_if proposer_address != global<stake::StakePool>(stake_pool).delegated_voter;
+
         include AbortsIfNotGovernanceConfig;
-        let current_time = timestamp::now_seconds();
+
+        /// get_voting_power(stake_pool)
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/aptos_governance.spec.move:152]: `///` --> `//` and could you give a bit more descriptive comment?

**Review by junkil-park — APPROVED** [review]
Thanks. The specs are very long. Could you use schemas to reduce the repeated spec as much as possible?

File: aptos-move/framework/aptos-framework/sources/aptos_governance.spec.move [hunk 2]
```diff
@@ -325,20 +522,82 @@ spec aptos_framework::aptos_governance {
...
+        let voting_forum = global<voting::VotingForum<GovernanceProposal>>(@aptos_framework);
         let proposal = table::spec_get(voting_forum.proposals, proposal_id);
-        requires chain_status::is_operating();
+        let post post_voting_forum = global<voting::VotingForum<GovernanceProposal>>(@aptos_framework);
+        let post post_proposal = table::spec_get(post_voting_forum.proposals, proposal_id);
+
+        aborts_if !string::spec_internal_check_utf8(voting::IS_MULTI_STEP_PROPOSAL_IN_EXECUTION_KEY);
+        let multi_step_in_execution_key = utf8(voting::IS_MULTI_STEP_PROPOSAL_IN_EXECUTION_KEY);
+        let post is_multi_step_proposal_in_execution_value = simple_map::spec_get(post_proposal.metadata, multi_step_in_execution_key);
+        // aborts_if simple_map::spec_contains_key(proposal.metadata, multi_step_in_execution_key) && !aptos_std::from_bcs::deserializable<vector<u8>>(b"true");
```
Comment (rahxephon89) [inline aptos-move/framework/aptos-framework/sources/aptos_governance.spec.move:538]: Do we need this commented-out aborts_if condition? If not, you can remove it

File: aptos-move/framework/aptos-framework/sources/aptos_governance.spec.move [hunk 3]
```diff
@@ -175,61 +216,202 @@ spec aptos_framework::aptos_governance {
...
-        // TODO: The variable `proposal_state` is the return value of the function `voting::get_proposal_state`.
-        // The calling level of `voting::get_proposal_state` is very deep,
-        // so the value of `proposal_state` cannot be obtained in the spec,
-        // and the `aborts_if` of `proposal_state` cannot be written.
-        // Can't cover all aborts_if conditions
-        pragma aborts_if_is_partial;
 
         requires chain_status::is_operating();
-        aborts_if !exists<ApprovedExecutionHashes>(@aptos_framework);
+        include Schema_add_approved_script_hash;
```
Comment (rahxephon89) [inline aptos-move/framework/aptos-framework/sources/aptos_governance.spec.move:348]: Could you change the schema name to something like `SchemaAddApprovedScriptHash`?

File: aptos-move/framework/aptos-framework/sources/aptos_governance.spec.move [hunk 4]
```diff
@@ -175,61 +216,202 @@ spec aptos_framework::aptos_governance {
...
-        // and the call level of the function `resolve` is too deep to obtain the parameter `bytes` of spec `from_bytes`,
-        // so verification cannot be performed.
-        // Can't cover all aborts_if conditions
-        pragma aborts_if_is_partial;
 
         requires chain_status::is_operating();
-        aborts_if !exists<voting::VotingForum<GovernanceProposal>>(@aptos_framework);
+
+        // verify voting::resolve
+        include Voting_Is_proposal_resolvable_Abortsif;
```
Comment (rahxephon89) [inline aptos-move/framework/aptos-framework/sources/aptos_governance.spec.move:387]: Same here, use upper camel case to rename the schema name.

**Review by rahxephon89 — CHANGES_REQUESTED** [review]
I am wondering whether the size of spec can be further reduced by introducing spec funcs for helper functions such get_voting_power.

**Comment by UIZorrot** [comment]
> I am wondering whether the size of spec can be further reduced by introducing spec funcs for helper functions such get_voting_power.

Well, we believe it's hard to reduce the size by helper function, since using opaque on these function always bringing new "function does not abort" issues. The other problem is solved.

**Review by rahxephon89 — APPROVED** [review]

**Pull request merged**